test/Dialect/Comb/comb-int-range-narrowing.mlir

```diff
@@ -0,0 +1,113 @@
+// RUN: circt-opt %s --comb-int-range-narrowing --canonicalize | FileCheck %s
+
+// CHECK-LABEL: @basic_csa
+hw.module @basic_csa(in %a : i1, in %b : i1, in %c : i1, out add_abc : i3) {
+// CHECK-NEXT: %false = hw.constant false
+// CHECK-NEXT: %0 = comb.concat %false, %a : i1, i1
```

Comment:
To make tests less fragile, we try to avoid hardcoding the SSA names of values into tests and use regex format and labelling instead - looking through existing tests should help figure this out, randomly picked example [here](https://github.com/llvm/circt/blob/main/test/Conversion/AIGToComb/aig-to-comb.mlir) - let me know if that's unclear at all!